torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,341 @@
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Tuple of lists that contain the cleaned strings for hypotheses and references
+
+    Raises:
+        ValueError: If a different language than 'en" or 'ja' is used
+        ValueError: If lenght of hypotheses is not equal to length of references
+    """
+    # sanity checks
+    if isinstance(hypotheses, str):
+        hypotheses = [hypotheses]
+    if isinstance(references, str):
+        references = [references]
+
+    if len(hypotheses) != len(references):
+        raise ValueError("Length of hypotheses must equal length of references")
+
+    # preprocess string
+    if language == "en":
+        preprocess_function = _preprocess_en
+    elif language == "ja":
+        preprocess_function = _preprocess_ja
+    else:
+        raise ValueError(f"Language {language} not supported, supported languages are 'en' and 'ja'")
+
+    hypotheses = [preprocess_function(hyp) for hyp in hypotheses]
+    references = [preprocess_function(ref) for ref in references]
+
+    return hypotheses, references
+
+
+def _eed_update(
+    hypotheses: Union[str, List[str]],
+    references: Union[str, List[str]],
+    language: Literal["en", "ja"] = "en",
+) -> Tuple[Tensor, Tensor]:
+    """Compute scores for EED.
+
+    Args:
+        hypotheses: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each input as a string or list of strings
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Tuple of scores and total sentences as floats
+    """
+    hypotheses, references = _preprocess_sentences(hypotheses, references, language)
+
+    # calculate score
+    scores = 0.0
+    total = 0.0
+
+    for hypothesis, reference in zip(hypotheses, references):
+        hyp: List[str] = list(hypothesis)
+        ref: List[str] = list(reference)
+        score = _eed_function(hyp, ref)
+        scores += score
+        total += 1.0
+
+    return tensor(scores), tensor(total)
+
+
+def eed(
+    hypotheses: Union[str, List[str]],
+    references: Union[str, List[str]],
+    language: Literal["en", "ja"] = "en",
+) -> Tensor:
+    """Computes extended edit distance score (`EED`_) [1] for strings or list of strings The metric utilises the
+    Levenshtein distance and extends it by adding an additional jump operation.
+
+    Args:
+        hypotheses: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each input as a string or list of strings
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Extended edit distance score as a tensor
+
+    Example:
+        >>> hypotheses = ["this is the prediction", "here is an other sample"]
```

Comment:
strictly not necessary, but it showcases to the user the recommended import path
```suggestion
        >>> from torchmetrics.functional import eed
        >>> hypotheses = ["this is the prediction", "here is an other sample"]
```

Reply:
will do